odl/tomo/backends/astra_setup.py

```diff
@@ -202,33 +202,40 @@ def astra_geom_to_vec(geometry):
         Numpy array of shape ``(number of angles, 12)``
     """
 
-    angles = geometry.angle_grid
-    num_angles = geometry.angle_grid.ntotal
+    angles = geometry.motion_grid
+    num_angles = geometry.motion_grid.ntotal
+
+    if isinstance(geometry, ConeBeamGeometry) and isinstance(geometry.detector, FlatDetector):
```

Comment:
Note: This is the only way to properly check that we have a flat panel cone beam geometry.